### PR TITLE
chore: drop kafka_skip_broken_messages from kafka_events_json

### PR DESCRIPTION
**Pull request by bretthoerner**
## Problem

We _shouldn't_ be sending any messages that don't deserialize properly, and so we'd like to go back to `kafka_skip_broken_messages = 0` in prod.

## Changes

Remake virtual Kafka table (and MV that's based on it) without the `kafka_skip_broken_messages` setting.

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

This is modeled after [a previous migration](https://github.com/PostHog/posthog/blob/1f8de249a91ba5726db98a52c4f7b8cd888e6d66/posthog/clickhouse/migrations/0047_add_insertion_timestamp_to_events.py) and should be reversible. Since it could pause the pipeline we should probably deploy early on a Monday and monitor it (with a revert PR ready).

### CODE REVIEW

**Comment by bretthoerner** [comment]
@fuziontech I'm not sure if I missed a tool to create these, I created the migration manually. I assume our migration script checks for the migration by name.

Speaking of which, there are two `0049` migrations here: https://github.com/PostHog/posthog/tree/master/posthog/clickhouse/migrations

**Review by tiina303 — APPROVED** [review]
lgtm

**Comment by bretthoerner** [comment]
Ping @fuziontech - looks like I need your approval.

**Review by fuziontech — APPROVED** [review]
lgtm! sorry for lag
🚢 it!

**Comment by fuziontech** [comment]
> @fuziontech I'm not sure if I missed a tool to create these, I created the migration manually. I assume our migration script checks for the migration by name.

No tool to make these, just manually. Checks for i+1 where i is the current migration id.

> Speaking of which, there are two 0049 migrations here. I hope that's OK/known?: https://github.com/PostHog/posthog/tree/master/posthog/clickhouse/migrations

oh, interesting. I would probably increment this to `0051` and bump the last migration to `0050`

**Review by fuziontech — CHANGES_REQUESTED** [review]
👀

**Comment by fuziontech** [comment]
Actually. Give me a second to put a PR to correct this. One of those migrations needs to be removed. One sec.

**Comment by fuziontech** [comment]
Once this lands, let's rebase this one and 🚢 it
https://github.com/PostHog/posthog/pull/19110

**Comment by bretthoerner** [comment]
Rebased 🤞

**Comment by bretthoerner** [comment]
Looks good, just needs your stamp @fuziontech

**Review by fuziontech — APPROVED** [review]
just kept rolling for all green tests.
Ship it!

**Comment by bretthoerner** [comment]
Actually going to close this in favor of the upcoming `_error` virtual columns.

This migration will allow bad messages to kill ingestion, where as the `_error` columns can be forwarded and stored in a table for us to analyze (without killing ingestion).

**Pull request closed without merging**
